code/add_aws_regions_to_database.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import Querying_Database as qdb
+
+from collections import defaultdict
+from geopy.distance import geodesic
+from math import inf
+from shapely import wkt, shortest_line, LineString, Point
+
+def find_closest_paths(lat: float, lon: float, db_path: str, max_distance: float):
+    querier = qdb.queryDatabase(db_path)
+    query = f"""SELECT * FROM standard_paths;"""
+    results = querier.execute_query(query)
+    
+    paths = []
+    for row in results:
+        fc = row[0]
+        fs = row[1]
+        fcc = row[2]
+        tc = row[3]
+        ts = row[4]
+        tcc = row[5]
+        dist_km = float(row[6])
+        path_wkt = row[7]
+        edge = ((fc, fs, fcc), (tc, ts, tcc))
+
+
+        linestring = wkt.loads(path_wkt)
+        # The linestring is an existing edge in the graph, and the point is the
+        # location of the cloud region. We get the shortest line between these
+        # two geometries, which gives us the shortest distance from a cloud
+        # region to an existing edge in the graph.
+        ps = shortest_line(Point(lon, lat), linestring).coords
+        # Since the WKT format is (lon, lat) and geopy.geodesic takes
+        # (lat, lon) pairs, we need to switch these.
+        p1 = (ps[0][1], ps[0][0])
+        p2 = (ps[1][1], ps[1][0])
+        distance = geodesic(p1, p2)
+        if distance.km < max_distance:
+            paths.append(row)
+
+    return paths
+
+# Taken from stackoverflow
```

Comment:
Possibly paste the url if you still remember it?

Reply:
Done